### DISCUSSION TITLE
Plugin for a language not supported by coc.nvim has show_documentation blocked by 'doHover'.

### DISCUSSION THREAD

**Discussion by vsandstrom** [neoclide/coc.nvim#3541] [Q&A] [answered]
Plugin for a language not supported by coc.nvim has its show_documentation blocked by 'doHover'.

When using 'K' to get documentation at cursor, this happens if I don't comment out coc.nvim settings and restart before working. 

Is there an easy way to disable this? I've tried to use conditional logic for filetypes, and I've searched both here and at the plugins issue thread and community, but cannot find anyone who has had this kind of issue, only some that sounds similar but isn't. The plugin in question is SCNvim. 

this is the error i get:
> [coc.nvim]: Error on “doHover”: current buffer 3 not attached

The easy way would be to map the 'documentation' to some other key, but it would be annoying. How would I suppress or disable the 'doHover' function in buffers of a certain filetype? 

The section in the coc.nvim settings where this is handled looks like this. 

    nnoremap <silent> K :call <SID>show_documentation()<CR>

    function! s:show_documentation()   
        if (index(['vim','help'], &filetype) >= 0) 
            execute 'h '.expand('<cword>')
        elseif (coc#rcp#ready())
            call CocActionAsync('doHover')
        else
            execute '!' . &keywordprog . " " . expand('<cword>')
        endif
    endfunction

**Comment by chemzqm** [comment, marked as answer]
Seems the buffer is not a normal buffer, check buftype by `:echo &buftype`.
Also checkout `CocAction('ensureDocument')`
